packages/tools/client-debugger/src/Registry.ts

```diff
@@ -45,6 +46,66 @@ export interface FluidClientDebuggerProps {
 	containerNickname?: string;
 }
 
+export interface DebuggerRegistryEvents extends IEvent {
+	/**
+	 * Emitted when a {@link FluidClientDebugger | clientDebugger} is registered.
+	 *
+	 * @eventProperty
+	 */
+	(event: "debuggerRegistered", listener: (containerId: string) => void): void;
+
+	/**
+	 * Emitted when a {@link FluidClientDebugger | clientDebugger} is closed.
+	 *
+	 * @eventProperty
+	 */
+	(event: "debuggerClosed", listener: (containerId: string) => void): void;
+}
+
+export class DebuggerRegistry extends IEventProvider<DebuggerRegistryEvents> {
+	private readonly registeredDebuggers: Map<string, FluidClientDebugger> = new Map();
+
+	constructor() {
+		super();
+	}
+
+	public initializeDebugger(props: FluidClientDebuggerProps): void {
+		const { containerId } = props;
+		const existingDebugger = this.registeredDebuggers.get(containerId);
+		if (existingDebugger !== undefined) {
+			console.warn(
+				`Active debugger registry already contains an entry for container ID "${containerId}". Override existing entry.`,
+			);
+			existingDebugger.dispose();
+		}
+
+		const clientDebugger = new FluidClientDebugger(props);
+		this.registeredDebuggers.set(containerId, clientDebugger);
+		this.emit("debuggerRegistered", containerId, clientDebugger);
+	}
+
+	public closeDebugger(containerId: string): void {
+		if (this.registeredDebuggers.has(containerId)) {
+			const clientDebugger = this.registeredDebuggers.get(containerId);
+			if (clientDebugger === undefined) {
+				console.warn(
+					`No active client debugger associated with container ID "${containerId}" was found.`,
+				);
+			} else {
+				clientDebugger.dispose();
+				this.registeredDebuggers.delete(containerId);
+				this.emit("debuggerClosed", containerId);
+			}
+		} else {
+			console.warn(`Fluid Client debugger never been registered.`);
+		}
+	}
+
+	public getRegisteredDebuggers(): Map<string, IFluidClientDebugger> {
```

Comment:
Rather than returning the underlying map, it might be nice to just return a readonly array of the current list. We can add methods to this type for getting a debugger by ID so that the external functions can still access them.

Reply:
Make sense! updated.